uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/FieldsPopupElementViewController.cpp

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ *  Copyright 2012-2024 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#include "FieldsPopupElementViewController.h"
+#include "PopupField.h"
+
+namespace Esri::ArcGISRuntime::Toolkit {
+
+/*!
+  \class Esri::ArcGISRuntime::Toolkit::FieldsPopupElementViewController
+  \inmodule ArcGISRuntimeToolkit
+  \ingroup ArcGISQtToolkitUiCppControllers
+  \brief In MVC architecture, this is the controller for the corresponding
+  \c FieldsPopupElementView.
+ */
+
+/*!
+ \brief Constructor
+ \list
+   \li \a parent Parent owning \c QObject.
+ \endlist
+ */
+FieldsPopupElementViewController::FieldsPopupElementViewController(QObject* parent)
+  : PopupElementViewItem{parent}
+{
+}
+
+/*!
+  \brief Destructor.
+  */
+FieldsPopupElementViewController::~FieldsPopupElementViewController() = default;
+
+/*!
+\brief Constructor. Takes a \a fieldsPopupElement and \a parent object.
+  */
+FieldsPopupElementViewController::FieldsPopupElementViewController(
+    QPointer<FieldsPopupElement> fieldsPopupElement, QObject* parent)
+  : PopupElementViewItem{QPointer<FieldsPopupElement>(std::move(fieldsPopupElement)), parent}
+{
+}
+
+/*!
+  \brief Returns the title of the \c FieldsPopupElement.
+ */
+QString FieldsPopupElementViewController::title() const
+{
+  return static_cast<FieldsPopupElement*>(popupElement())->title();
+}
+
+/*!
+  \brief Returns the combination of each element from \l {Esri::ArcGISRuntime::FieldsPopupElement::labels} {labels} and
+  \l {Esri::ArcGISRuntime::FieldsPopupElement::formattedValues} {formattedValues} of the
+  \c FieldsPopupElement in a QList of QMap elements
+  */
+QVariantList FieldsPopupElementViewController::values()
```

Comment:
If this is something we expect most popup implementations would need, then we should consider if this is something we should put in the API itself

Reply:
I think a ListModel with these accessible through roles would make a lot of sense to me since I imagine it would be really common to access both the label and it's value in the same context.

Reply:
I'll elicit feedback from the team during the review.

Reply:
>in a QList of QMap elements

This sounds more like a list model, if I understand the data layout correctly.

Reply:
That's what I was trying to accomplish without going as far as subclassing QAbstractListModel just for this scenario because I thought this could be a useful addition to the API.